agent/agent.go

```diff
@@ -92,6 +99,77 @@ func (agent *Agent) RegisterServer() error {
 	return nil
 }
 
+func (agent *Agent) PerformUpgrade() error {
+	package_list, err := agent.client.FetchUpgradeablePackages()
+
+	if err != nil {
+		log.Info("Can't fetch upgrade info: %s", err)
+		return err
+	}
+
+	if agent.server.DebianLike() {
+		agent.runDebianUpgrade(package_list)
+	} else {
+		log.Fatal("Sorry, we don't support your operating system at the moment. Is this a mistake? Run `appcanary detect-os` and tell us about it at support@appcanary.com")
+	}
+	return nil
+}
+
+func (agent *Agent) runDebianUpgrade(package_list map[string]string) {
+	updateCmd := "apt-get"
+	updateArg := []string{"update", "-q"}
+
+	installCmd := "apt-get"
+	installArg := []string{"install", "--only-upgrade", "--no-install-recommends", "-y", "-q"}
+
+	for name, version := range package_list {
+		installArg = append(installArg, name+"="+version)
+	}
+
+	if env.DryRun {
+		log.Info("Running upgrade in dry-run mode...")
+	}
+
+	err := agent.runCmd(updateCmd, updateArg)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = agent.runCmd(installCmd, installArg)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func (agent *Agent) runCmd(cmd_name string, args []string) error {
+	_, err := exec.LookPath(cmd_name)
+
+	if err != nil {
+		log.Fatal("Can't find " + cmd_name)
+	}
+
+	cmd := exec.Command(cmd_name, args...)
+
+	var output bytes.Buffer
+	cmd.Stdout = &output
+	cmd.Stderr = &output
+
+	log.Infof("Running: %s %s", cmd_name, strings.Join(args, " "))
```

Comment:
Cut the "Running".

Dry run should be copy-and-pastable, so you can take the output of dry run and just paste it into a terminal or chef script or w/e